queries.sql

```diff
@@ -11,4 +11,49 @@ SELECT (name, secape_attempts) FROM animals WHERE weight_kg > 10.5;
 
 SELECT * FROM animals WHERE neutered = TRUE;
 
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 and 17.3;
+
+-- Defining the species
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name = '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = NULL;
+SELECT * FROM animals;
+COMMIT;
+
+-- deleting and ROLLBACK
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+
+-- DELETE ANIMALS WITH CONDITIONS
+BEGIN;
+DELETE  FROM animals
+WHERE date_of_birth > '2022-01-01';
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg *-1;
+ROLLBACK;
```

Comment:
- [ ] This ROLLBACK should be to the savepoint you must have created above.
`ROLLBACK TO SP1;`

Reply:
well resolved! ✔️